tests/test_sklearn_models.py

```diff
@@ -1,118 +1,32 @@
-
-            tiledb_sklearn_obj = SklearnTileDB(uri=tiledb_array)
-            tiledb_sklearn_obj.save(model=model)
-
-            loaded_model = tiledb_sklearn_obj.load()
-
-            model_pred = model.predict(iris.data)
-            loaded_model_pred = loaded_model.predict(iris.data)
-
-            self.assertTrue(np.array_equal(model_pred, loaded_model_pred))
-
-    def test_save_load_decision_tree_regressor(self):
-        with tempfile.TemporaryDirectory() as tiledb_array:
-            model = DecisionTreeRegressor()
-            model.fit(iris.data, iris.target)
-
-            tiledb_sklearn_obj = SklearnTileDB(uri=tiledb_array)
-            tiledb_sklearn_obj.save(model=model)
-
-            loaded_model = tiledb_sklearn_obj.load()
-
-            model_pred = model.predict(iris.data)
-            loaded_model_pred = loaded_model.predict(iris.data)
-
-            self.assertTrue(np.array_equal(model_pred, loaded_model_pred))
-
-    def test_save_load_decision_tree_classifier(self):
-        with tempfile.TemporaryDirectory() as tiledb_array:
-            model = DecisionTreeClassifier()
-            model.fit(iris.data, iris.target)
-
-            tiledb_sklearn_obj = SklearnTileDB(uri=tiledb_array)
-            tiledb_sklearn_obj.save(model=model)
-
-            loaded_model = tiledb_sklearn_obj.load()
-
-            model_pred = model.predict(iris.data)
-            loaded_model_pred = loaded_model.predict(iris.data)
-
-            self.assertTrue(np.array_equal(model_pred, loaded_model_pred))
-
-    def test_save_load_with_update(self):
-        with tempfile.TemporaryDirectory() as tiledb_array:
-            model = LinearRegression()
-            model.fit(iris.data, iris.target)
-
-            tiledb_sklearn_obj = SklearnTileDB(uri=tiledb_array)
-            tiledb_sklearn_obj.save(model=model)
-
-            loaded_model = tiledb_sklearn_obj.load()
-
-            model_pred = model.predict(iris.data)
-            loaded_model_pred = loaded_model.predict(iris.data)
-
-            self.assertTrue(np.array_equal(model_pred, loaded_model_pred))
-
-            model = ElasticNet()
-            model.fit(iris.data, iris.target)
-
-            tiledb_sklearn_obj = SklearnTileDB(uri=tiledb_array)
-            tiledb_sklearn_obj.save(model=model, update=True)
-
-            loaded_model = tiledb_sklearn_obj.load()
-
-            model_pred = model.predict(iris.data)
-            loaded_model_pred = loaded_model.predict(iris.data)
-
-            self.assertTrue(np.array_equal(model_pred, loaded_model_pred))
+tested_modules = ["svm", "linear_model", "naive_bayes", "tree"]
+
+
+def model_explorer():
+    model_list = []
+    all_estim = sklearn.utils.all_estimators()
+    for estim_name, estim_class in all_estim:
+        if str(estim_class).split(".")[1] in tested_modules:
+            model_list.append(estim_class)
+    return model_list
+
+
+@pytest.mark.parametrize(
+    "net",
+    [
+        *model_explorer(),
```

Comment:
This can now be just `model_explorer()` (or `list(iter_models("svm", "linear_model", "naive_bayes", "tree"))` with the previous comment change).